### PR TITLE
Changes as discussed

### PR DESCRIPTION
**Pull request by mschottdorf**
More specifically:
(1) assertions are back in the code to enforce the API
(2) pytest-code is refactored. I.e. is removed from `common` and `controller`, and exclusively contained in /tests.
(3) A more elegant way to wait for HAL-Timeout in the controller using the `@timeout`-decorator

### CODE REVIEW

File: pvp/common/loggers.py [hunk 1]
```diff
@@ -269,13 +280,33 @@ def _open_logfile(self):
...
             self.derived_table = self.h5file.root.derived_quantities.readout
 
+        if "/program_information" not in self.h5file:
+            self.logger.info('Generating /program_information table in: ' + self.file )
+            group = self.h5file.create_group("/", 'program_information', 'General information about PVP-1')
+            self.program_table = self.h5file.create_table(group, 'readout', ProgramData, "Program information",
+                                                          filters = pytb.Filters(
+                                                              complevel=self.compression_level,
+                                                              complib='zlib'),
+                                                          expectedrows=1000000)
```
Comment (sneakers-the-rat) [inline pvp/common/loggers.py:290]: the infinite future of pvp upgrades ;)

File: pvp/common/loggers.py [hunk 2]
```diff
@@ -420,7 +451,10 @@ def load_file(self, filename = None):
             table = file.root.derived_quantities.readout
             derived_data = table.read()
 
-        data_dict = {"waveform_data": waveform_data, "control_data": control_data, "derived_data": derived_data}
+            table = file.root.program_information.readout
+            program_data = table.read()
+
+        data_dict = {"waveform_data": waveform_data, "control_data": control_data, "derived_data": derived_data, "program_information": program_data}
```
Comment (sneakers-the-rat) [inline pvp/common/loggers.py:457]: does this save the version in every row of the continuous data?

File: pvp/gui/widgets/control_panel.py [hunk 3]
```diff
@@ -98,15 +99,7 @@ def init_ui(self):
...
-        version = pvp.__version__
-        try:
-            # get git version
-            git_version = subprocess.check_output(['git', 'describe', '--always'],
-                                                  cwd=os.path.dirname(__file__)).strip().decode('utf-8')
-            version = " - ".join([version, git_version])
-        except Exception: # pragma: no cover
-            # no problem, just use package version
-            pass
+        version = get_version()
```
Comment (sneakers-the-rat) [inline pvp/gui/widgets/control_panel.py:102]: thanks for cleaning this up for me

**Review by sneakers-the-rat — APPROVED** [review]
looks good to me ;) thanks for the cleanup

File: pvp/common/loggers.py (hunk 2 above)
Comment (mschottdorf) [reply pvp/common/loggers.py:457]: no, it's saved only once at the beginning

File: pvp/common/loggers.py (hunk 1 above)
Comment (mschottdorf) [reply pvp/common/loggers.py:290]: yeah, could be shorter 👍

**Pull request merged**